### PR TITLE
`waitForVMCommandExecution`  doesn't exit loop on `PENDING` or `IN_PROGRESS` states

### PR DESCRIPTION
**Pull request by Sladuca**
# Description

Problem: `waitForVMCommandExecution` rejects after the first loop due to the `finally` block here: https://github.com/privacy-scaling-explorations/p0tion/blob/2249f7fb82e3c0f0fc42a712ada7a08399bbc567/packages/backend/src/functions/circuit.ts#L261-L265

Solution: `waitForVMCommandExecution` exhaustively checks every possible command status and explicitly retries in the `PENDING` and `IN_PROGRESS` states.

Note: this PR is a lot bigger than it should be because I ran `prettier:fix`. I added three comments below flagging the relevant changes.

# Checklist:

-   [X] My code follows the style guidelines of this project
-   [X] I have performed a self-review of my own code
-   [X] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have made corresponding changes to the documentation
-   [X] My changes generate no new warnings
-   [X] Any dependent changes have been merged and published in downstream modules
-   [X] I reviewed the [code of conduct](https://github.com/privacy-scaling-explorations/p0tion/blob/main/CODE_OF_CONDUCT.md) and [contributors' guide](https://github.com/privacy-scaling-explorations/p0tion/blob/main/CONTRIBUTING.md)

### CODE REVIEW

File: packages/backend/src/functions/circuit.ts [hunk 1]
```diff
@@ -214,57 +215,71 @@ const coordinate = async (
...
-            } else if (cmdStatus === CommandInvocationStatus.CANCELLED) {
-                logAndThrowError(SPECIFIC_ERRORS.SE_VM_CANCELLED_COMMAND_EXECUTION)
-                reject()
-            } else if (cmdStatus === CommandInvocationStatus.DELAYED) {
-                logAndThrowError(SPECIFIC_ERRORS.SE_VM_DELAYED_COMMAND_EXECUTION)
-                reject()
-            }
-        } catch (error: any) {
-            printLog(`Invalid command ${commandId} execution`, LogLevel.DEBUG)
+const waitForVMCommandExecution = (ssm: SSMClient, vmInstanceId: string, commandId: string): Promise<void> =>
```
Comment (Sladuca) [inline packages/backend/src/functions/circuit.ts:223]: relevant change 1

Comment (Sladuca) [inline packages/backend/src/functions/circuit.ts:223]: relevant change 1

File: packages/backend/src/functions/circuit.ts [hunk 2]
```diff
@@ -783,9 +792,7 @@ export const verifycontribution = functionsV2.https.onCall(
                 printLog(`Starting the execution of command ${commandId}`, LogLevel.DEBUG)
 
                 // Step (1.A.3.3).
-                return new Promise<void>((resolve, reject) =>
-                    waitForVMCommandExecution(resolve, reject, ssm, vmInstanceId, commandId)
-                )
+                return waitForVMCommandExecution(ssm, vmInstanceId, commandId)
```
Comment (Sladuca) [inline packages/backend/src/functions/circuit.ts:795]: relevant change 2

**Review by ctrlc03 — APPROVED** [review]
LGTM! great catch there

**Pull request merged**